### PR TITLE
- work on LVM thin provisioning

### PR DESCRIPTION
**Pull request by aschnell**
For https://trello.com/c/m2tWJPYI/724-5-libstorage-ng-lvm-thin-provisioning-performing-operations.

### CODE REVIEW

File: testsuite/Utils/math.cc [hunk 1]
```diff
@@ -21,3 +21,28 @@ BOOST_AUTO_TEST_CASE(test_is_power_of_two)
...
+
+
+BOOST_AUTO_TEST_CASE(test_next_power_of_two)
+{
+    BOOST_CHECK_EQUAL(next_power_of_two(1), 1);
+
+    BOOST_CHECK_EQUAL(next_power_of_two(2), 2);
+
+    BOOST_CHECK_EQUAL(next_power_of_two(3), 4);
+    BOOST_CHECK_EQUAL(next_power_of_two(3), 4);
```
Comment (wfeldt) [inline testsuite/Utils/math.cc]: Why check it twice?

Comment (aschnell) [reply testsuite/Utils/math.cc]: It cannot hurt :-) No, likely ```BOOST_CHECK_EQUAL(next_power_of_two(4), 4);``` was intended.

File: storage/Devices/LvmVgImpl.cc [hunk 2]
```diff
@@ -201,6 +201,41 @@ namespace storage
...
+
+		unsigned long long chunk_size = LvmLv::Impl::default_chunk_size(data_size);
+		unsigned long long metadata_size =
+		    LvmLv::Impl::default_metadata_size(data_size, chunk_size, extent_size);
+
+		// Subtract metadata size twice due to spare metadata. This is
+		// a bit conservative since there might already be a spare
+		// metadata.
+
+		return data_size - 2 * metadata_size;
```
Comment (wfeldt) [inline storage/Devices/LvmVgImpl.cc]: Can this get negative in some corner case?

File: doc/lvm.md [hunk 3]
```diff
@@ -45,3 +46,16 @@ Solutions:
 
 Solution 2 is implemented.
 
+
+Creating a Logical Volume
+-------------------------
+
+When creating a new logical volume the size should not be bigger than the
+value which max_size_for_lvm_lv() returned. A calculation simple based on
```
Comment (wfeldt) [inline doc/lvm.md]: simply?

File: storage/Devices/LvmVgImpl.cc (hunk 2 above)
Comment (aschnell) [reply storage/Devices/LvmVgImpl.cc]: Well, yes. I'll add a check.

File: storage/Utils/Math.cc [hunk 4]
```diff
@@ -50,10 +50,18 @@ namespace storage
 
 
     unsigned long long
+    round_down(unsigned long long i, unsigned long long m)
+    {
+	unsigned long long r = i % m;
+	return r == 0 ? i : i - r;
```
Comment (wfeldt) [inline storage/Utils/Math.cc]: simply `return i - r`?

**Comment by wfeldt** [comment]
lgtm; just some minor comments

**Pull request merged**